docs/how-to-guides/how-to-test-contract-with-clarinet-sdk.md

```diff
@@ -0,0 +1,516 @@
+- [Costs optimization](#costs-optimization)
+- [Produce both coverage and costs reports.](#produce-both-coverage-and-costs-reports)
+- [Run tests in CI](#run-tests-in-ci)
+- [Custom Vitest matchers](#custom-vitest-matchers)
+  - [Check clarity type](#check-clarity-type)
+  - [Response type](#response-type)
+  - [Optional type](#optional-type)
+  - [Simple clarity types](#simple-clarity-types)
+  - [Other composite types](#other-composite-types)
+
+## Requirements
+
+The SDK requires Node.js >= 18.0 and NPM to be installed. [Volta](https://volta.sh/) is a great tool to install and manage JS tooling.
+
+To follow this tutorial, you must have the Clarinet CLI installed as well.
+
+## Set up the Clarity contract and unit tests
+
+Let us consider a `counter` smart contract to understand how to write unit tests for our application requirements.
+
+Create a new Clarinet project with a `counter` contract.
+
+```console
+clarinet new counter
+cd counter
+clarinet contract new counter
+```
+
+And this will be the content of our smart contract.
+It keeps track of an initialized value, allows for incrementing and decrementing, and prints actions as a log.
+
+```clarity
+;; counter.clar
+(define-data-var count uint u1)
+
+(define-public (increment (step uint))
+  (let ((new-val (+ step (var-get count)))) 
+    (var-set count new-val)
+    (print { object: "count", action: "incremented", value: new-val })
+    (ok new-val)
+  )
+)
+
+(define-public (decrement (step uint))
+  (let ((new-val (- step (var-get count)))) 
+    (var-set count new-val)
+    (print { object: "count", action: "decremented", value: new-val })
+    (ok new-val)
+  )
+)
+
+(define-read-only (read-count)
+  (ok (var-get count))
+)
+```
+
+### Migrating between Clarinet 1 and Clarinet 2
+
+> Note: Clarinet 2 will be released in October 2023, and will create the right boilerplate files. But if a project has been created with Clarinet 1, this prepare to project to run the SDK and Vitest.
+
+Executing this script in a Clarinet 1 project will initialise NPM and Vitest. It will also create a sample test file.
+
+```console
+npx @hirosystems/clarinet-sdk@latest
+```
+
+This script wil lask you if you want to run npm install now, you can press enter to do so.
+It can take a few seconds.
+
+The file `tests/counter_test.ts` that was created by `clarinet contract new counter` can be deleted.  
+
+You can have a look at `tests/contract.test.ts`, it's a sample file showing how to use the SDK with Vitest.
+It can safely be deleted.
+
+
+### Unit tests for `counter` example
+
+Create a file `tests/counter.test.ts` with the following content:
+
+```ts
+import { Cl } from "@stacks/transactions";
+import { describe, expect, it } from "vitest";
+
+const accounts = vm.getAccounts();
+const address1 = accounts.get("wallet_1")!;
+
+describe("test increment method", () => {
+  it("increments the count by the given value", () => {
+    const res1 = vm.callPublicFn("counter", "increment", [Cl.uint(1)], address1);
```

Comment:
Nitpick: Can we expand all the way to `result1`?

Reply:
Sure, expanded allll the way to `incrementResponse`